invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/search_app_customizations.js

```diff
@@ -5,83 +5,36 @@
 // Invenio App RDM is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
 
-import React from "react";
 import { overrideStore } from "react-overridable";
-import { Card, Input, Item } from "semantic-ui-react";
 import _ from "lodash";
 import _truncate from "lodash/truncate";
-
-const RDMRecordResultsListItem = ({ result, index }) => {
-  const description = _.get(
-    result,
-    "metadata.descriptions[0].description",
-    "No description"
-  );
-
-  return (
-    <Item key={index} href={`/records/${result.id}`}>
-      <Item.Content>
-        <Item.Header>{result.metadata.titles[0].title}</Item.Header>
-        <Item.Description>
-          {_truncate(description, { length: 200 })}
-        </Item.Description>
-      </Item.Content>
-    </Item>
-  );
-};
-
-overrideStore.add("ResultsList.item", RDMRecordResultsListItem);
-
-const RDMRecordResultsGridItem = ({ result, index }) => {
-  const description = _.get(
-    result,
-    "metadata.descriptions[0].description",
-    "No description"
-  );
-  return (
-    <Card fluid key={index} href={`/records/${result.id}`}>
-      <Card.Content>
-        <Card.Header>{result.metadata.titles[0].title}</Card.Header>
-        <Card.Description>
-          {_truncate(description, { length: 200 })}
-        </Card.Description>
-      </Card.Content>
-    </Card>
-  );
-};
-
-overrideStore.add("ResultsGrid.item", RDMRecordResultsGridItem);
-
-const RDMRecordSearchBarElement = ({
-  placeholder: passedPlaceholder,
-  queryString,
-  onInputChange,
-  executeSearch,
-}) => {
-  const placeholder = passedPlaceholder || "Search";
-  const onBtnSearchClick = () => {
-    executeSearch();
-  };
-  const onKeyPress = (event) => {
-    if (event.key === "Enter") {
-      executeSearch();
-    }
-  };
-  return (
-    <Input
-      action={{
-        icon: "search",
-        onClick: onBtnSearchClick,
-        className: "search",
-      }}
-      placeholder={placeholder}
-      onChange={(event, { value }) => {
-        onInputChange(value);
-      }}
-      value={queryString}
-      onKeyPress={onKeyPress}
-    />
-  );
+import {
+  RDMRecordResultsListItem,
+  RDMRecordResultsGridItem,
+  RDMRecordSearchBarElement,
+} from "./search";
+
+async function registerComponent(componentId, defaultComponent) {
+  let component;
+  try {
+    const module = await import(
+      /* webpackMode: "eager" */
+      `../../templates/search/${componentId}.jsx`
+    );
+    component = module.default;
+  } catch (error) {
+    component = defaultComponent;
+  } finally {
+    overrideStore.add(componentId, component);
+  }
+}
+
+const componentMapping = {
+  "ResultsList.item": RDMRecordResultsListItem,
+  "ResultsGrid.item": RDMRecordResultsGridItem,
+  "SearchBar.element": RDMRecordSearchBarElement,
```

Comment:
**Minor:** I find myself also in the need of defining such a listing to define default components in `invenio-search-ui`... Probably a full listing can start in `invenio-search-ui`, and then later get overwritten on the "module-level" by e.g. `invenio-app-rdm`, `invenio-communities`, etc. for the various search apps.